### PR TITLE
schema: allow refresh-mode and stop-mode

### PR DESCRIPTION
**Pull request by sergiusens**
Allow and validate `refresh-mode` and `stop-mode`.

- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] If this is a bugfix. Have you checked that there is a bug report open for the issue you are trying to fix on [bug reports](https://bugs.launchpad.net/snapcraft)?
- [x] If this is a new feature. Have you discussed the design on the [forum](https://forum.snapcraft.io)?
- [x] Have you successfully run `./runtests.sh static`?
- [x] Have you successfully run `./runtests.sh unit`?

-----

### CODE REVIEW

File: schema/snapcraft.yaml [hunk 1]
```diff
@@ -253,6 +253,24 @@ properties:
               - forking
               - oneshot
               - notify
+          refresh-mode:
+              type: string
+              description: controls if the app should be restarted at all
+              enum:
+                - endure
+                - restart
+          stop-mode:
```
Comment (kyrofa) [inline schema/snapcraft.yaml]: Both `refresh-mode` and `stop-mode` can only be used for daemons. You can probably use dependencies for this to validate in the schema.

File: tests/unit/project_loader/test_config.py [hunk 2]
```diff
@@ -2598,6 +2598,68 @@ def test_valid_restart_conditions(self):
         project_loader.Validator(self.data).validate()
 
 
+class ValidStopRefreshModesTestCase(ValidationBaseTestCase):
```
Comment (kyrofa) [inline tests/unit/project_loader/test_config.py]: Can we split these into separate test cases, please (stop and refresh)?

File: tests/unit/test_meta.py [hunk 3]
```diff
@@ -500,6 +500,46 @@ def test_create_meta_with_hook(self):
             "Expected generated 'bar' hook to not contain 'plugs'")
 
 
+class RefreshStopTestCase(CreateBaseTestCase):
```
Comment (kyrofa) [inline tests/unit/test_meta.py]: Also please split this into stop and refresh cases.

File: tests/unit/project_loader/test_config.py (hunk 2 above)
Comment (sergiusens) [reply tests/unit/project_loader/test_config.py]: awe, but I was so smart here

File: schema/snapcraft.yaml [hunk 4]
```diff
@@ -218,9 +218,11 @@ properties:
...
           - command
         dependencies:
-          stop-command: ["daemon"]
-          post-stop-command: ["daemon"]
-          reload-command: ["daemon"]
+          refresh-mode: [daemon]
+          stop-mode: [daemon]
+          stop-command: [daemon]
+          post-stop-command: [daemon]
+          reload-command: [daemon]
```
Comment (kyrofa) [inline schema/snapcraft.yaml]: Very nice. Not the BEST error in the world, but it's not bad.

**Pull request merged**